src/hooks/useIsUserVenueOwner.ts

```diff
@@ -0,0 +1,19 @@
+import { parentVenueSelector } from "utils/selectors";
+import { useConnectCurrentVenueNG } from "./useConnectCurrentVenueNG";
+import { useSelector } from "./useSelector";
+import { useUser } from "./useUser";
+import { useVenueId } from "./useVenueId";
+
+export const useIsUserVenueOwner = (): boolean => {
+  const { user } = useUser();
+
+  const currentVenueId = useVenueId();
+  const { currentVenue } = useConnectCurrentVenueNG(currentVenueId);
+  const parentVenue = useSelector(parentVenueSelector);
+
+  if (!currentVenue || !user) return false;
+
+  const venueOwners = parentVenue ? parentVenue.owners : currentVenue.owners;
+
+  return venueOwners.includes(user.uid);
```

Comment:
The logic here should be:

- check `currentVenue.owners.includes(user.uid)`
- if no match, and there is a `currentVenue.parentVenueId`, then load that as `parentVenue` and check `parentVenue.owners.includes(user.uid)`
- if no match, and there is a parentVenue.parentVenueId`, then repeat the above process of checking/loading the parent venue/etc until there is no more `parentVenueId` (at which point you have reached the 'root parent venue' aka 'sovereign venue')
- if there is no match on the 'root parent venue' aka 'sovereign venue', then we can finally consider that there is no match, and that the user doesn't have permission

Obviously we want to ensure we do this in an efficient way, so that it only completes the 're-checks' when required, and properly caches any data otherwise.